pontoon/sync/core/translations_from_repo.py

```diff
@@ -433,10 +435,31 @@ def update_db_translations(
             f"[{project.slug}] Created {str_n_translations(created)} from repo changes"
         )
 
+    badge_update = ("", 0)
     if actions:
+        translation_before_level = log_user.badges_translation_level
+        review_before_level = log_user.badges_review_level
+
         ActionLog.objects.bulk_create(actions)
 
-    return created, list(suggestions.values())
+        if (
+            log_user.badges_translation_level > translation_before_level
+            and log_user.username != "pontoon-sync"
+        ):
+            send_badge_notification(
+                log_user, "Translation Champion", log_user.badges_translation_level
+            )
+            badge_update = ("Translation Champion", log_user.badges_translation_level)
```

Comment:
The following would honor the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle:
```suggestion
            badge = "Translation Champion"
            level = log_user.badges_translation_level
            send_badge_notification(log_user, badge, level)
            badge_update = (badge, level)
```